src/bplus_tree.rs

```diff
@@ -0,0 +1,525 @@
+use crate::chunk_pointer::{ChunkHandler, ChunkPointer};
+use std::{
+    fs::File,
+    io::{self, ErrorKind},
+    os::unix::fs::FileExt,
+    path::PathBuf,
+};
+extern crate chunkfs;
+use chunkfs::{Data, DataContainer, Database};
+
+const MAXSIZE: u64 = 2 << 20;
+
+pub struct BPlus<K> {
+    t: usize,
+    root: Option<Box<Node<K>>>,
+    path: PathBuf,
+    file_number: usize,
+    offset: u64,
+    current_file: File,
+}
+
+#[derive(Clone)]
+#[allow(dead_code)]
+struct Node<K> {
+    leaf: bool,
+    key_num: usize,
+    keys: Vec<K>,
+    children: Vec<Option<Box<Node<K>>>>,
+    pointers: Vec<ChunkHandler>,
+}
+
+impl<K: Ord + Clone + Default> Node<K> {
+    fn new(leaf: bool) -> Self {
+        let new_node = Node {
+            leaf: leaf,
+            key_num: 0,
+            keys: Vec::new(),
+            children: Vec::new(),
+            pointers: Vec::new(),
+        };
+        new_node
+    }
+}
+#[allow(dead_code)]
+impl<K: Ord + Clone + Default> BPlus<K> {
+    pub fn new(t: usize, path: PathBuf) -> Self {
+        let root = Node::new(true);
+        let mut nodes: Vec<Box<Node<K>>> = Vec::new();
+        nodes.push(Box::new(root));
+        let path_to_file = path.join("0");
+        let res = File::create(path_to_file);
+        let current_file;
+        match res {
+            Ok(x) => current_file = x,
+            Err(_x) => panic!(),
+        }
+        Self {
+            t,
+            root: Some(Box::new(Node::new(true))),
+            path,
+            file_number: 0,
+            offset: 0,
+            current_file,
+        }
+    }
+
+    fn find_leaf<'a>(&self, node: &'a Box<Node<K>>, key: &K) -> Option<&'a Box<Node<K>>> {
+        let i;
+        let res = node.keys.binary_search(key);
+        match res {
+            Ok(x) => i = x + 1,
+            Err(x) => i = x,
+        }
+        if node.leaf {
+            for i in 0..node.key_num {
+                if node.keys[i] == *key {
+                    return Some(node);
+                }
+            }
+            return None;
```

Comment:
```suggestion
        if node.keys.contains(key) {
                Some(node)
            } else {
                None
            }
```
почаще используй итераторы, они делают код лаконичнее, красивее и более легко читаемым